lib/active_merchant/billing/gateways/plexo.rb

```diff
@@ -208,10 +219,21 @@ def get_authorization_from_url(url)
         url.split('/')[1]
       end
 
+      def reorder_amount_fields(response)
+        return response unless response['amount']
+
+        amount_obj = response['amount']
+        response['amount'] = amount_obj['total'].to_i if amount_obj['total']
+        response['currency'] = amount_obj['currency'] if amount_obj['currency']
+        response['amount_details'] = amount_obj['details'] if amount_obj['details']
+        response
+      end
+
       def commit(action, parameters)
         base_url = (test? ? test_url : live_url)
         url = build_url(action, base_url)
         response = parse(ssl_post(url, parameters.to_json, header(parameters)))
+        response = reorder_amount_fields(response) if action == 'authonly'
```

Comment:
Why are we doing this reordering, but only on an `authonly` transaction?

If this is necessary, I think it is worth including something in the tests showing that `authonly` transactions return responses with the amount fields in the expected format.

Reply:
Discussed and I have added unit test to demonstrate the use case.